File: DSA/stack/java/Stack.java
package stack.java;

public class Stack {
    private int arr[];
    private int capacity;
    private int topIndex;
    Stack(int x){
        this.capacity=x;
        this.topIndex=-1;
        this.arr=new int[x];
    }
    public int size(){
        return this.topIndex+1;
    }
    public int peek(){
        if(topIndex==-1){
            System.out.println("No element in the stack");
            return -1;
        }
        return arr[topIndex];
    }
    public void pop() {
        if(topIndex==-1){
            System.out.println("No element in the stack");
            return;
        }
        if (isEmpty()) {
            // throw new IllegalStateException("Stack is empty, cannot pop.");
            return;
        }
        topIndex--;
    }
    
    public void push(int p){
        topIndex++;
        arr[topIndex]=p;
    }
    public boolean isEmpty(){
        return topIndex==-1;
    }

}

File: DSA/stack/cpp/Stack.cpp
#include <bitset>
#include <algorithm>
#include <cassert>
#include <cctype>
#include <cmath>
#include <complex>
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include <ctime>
#include <deque>
#include <functional>
#include <iomanip>
#include <iostream>
#include <iterator>
#include <limits>
#include <list>
#include <map>
#include <numeric>
#include <queue>
#include <set>
// #include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>
#include <fstream>
#include <sstream>
using namespace std;
class Stack{
    private:
    int topIndex;
    int capacity;
    int *arr;
    public:
   Stack(int x){
    arr=new int[x];
    this->capacity=x;
    this->topIndex=-1;
   }
  int size(){
    return this->topIndex+1;
   }
   int pop(){
    topIndex--;
    return arr[topIndex];
   }
   void push(int x){
    topIndex++;
    arr[topIndex]=x;
   }
   bool isEmpty(){
    return this->topIndex==-1;
   }
   int peek(){
    return arr[this->topIndex];
   }
};
signed main(){
Stack st=Stack(5);
st.push(6);
cout<<st.peek();
return 0;
}